### PR TITLE
Enable cross-platform use with .NET Core

### PR DESCRIPTION
**Pull request by rikedyp**
Set ⎕USING based on .NET version
Data is now read using .NET stream reader
Tested on Windows and Linux

### CODE REVIEW

**Comment by rikedyp** [comment]
Now actually traps errors, and uses ⎕SIGNAL properly (errors are available in ⎕DMX)

**Comment by PaulMansour** [comment]
Hi Richard. Just saw this. I'm embarrassed to say I have never dealt with a merge request before! Thanks for contributing. I'll look over the code closely over the weekend or on Monday. One quick question/comment.   `ReadZipFile` has been drastically simplified, which is nice. The code that you deleted was cooked up by my co-worker Norbert, maybe with some help with John Daintree. This was necessary because a direct approach simply failed (or took forever) on a file of any decent size. I can't remember what the direct approach was, maybe it was different than what you have in the pull request.   Did you test this on large files  using both existing .NET and new .NET Core?

File: APLSource/Main/GetWorkbookInfo.aplf [hunk 1]
```diff
@@ -1,6 +1,8 @@
  GetWorkbookInfo←{
      ⍝ ⍵ ←→ FQP of .xlsx file
      ⍝   ←→ Space of usefull stuff
+     0::⎕SIGNAL⊂⎕DMX.(('EN'EN)('Message'Message))
```
[the remark this answers is not in the document]
Comment (rikedyp) [reply APLSource/Main/GetWorkbookInfo.aplf]: Yep good point - to what extent this is a "developer package" or and end user thing I didn't know. I also clearly didn't know how to do error handling properly. For now I will simply remove these.

File: APLSource/Main/UsingDotNetFileSystem.aplf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ UsingDotNetFileSystem←{
+⍝ Return ⎕USING value depending on available .NET version
+     ⎕IO←1
```
[the remark this answers is not in the document]
Comment (rikedyp) [reply APLSource/Main/UsingDotNetFileSystem.aplf]: Nope. Removed.

File: APLSource/Main/GetWorkbookInfo.aplf [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+     3 6::⎕SIGNAL⊂('EN' 16)('Message' '.NET unavailable')
```
[the remark this answers is not in the document]
Comment (rikedyp) [reply APLSource/Main/GetWorkbookInfo.aplf]: Yep. Removed.

**Comment by rikedyp** [comment]
Re: testing on large files

`ReadData` simply does `GetWorkbookInfo` and `GetSheetData`

Both `c` (source at head of this repository) and `rp` (head of rikedyp) seem to perform the same within the tolerance of `]runtime`

## big74.xlsx is a 16.6MB file
```
      ]runtime -c "c.XL2APL.APLSource ReadData '/tmp/big74.xlsx'" "rp.APLSource ReadData'/tmp/big74.xlsx'"
                                                                                                        
  c.XL2APL.APLSource ReadData '/tmp/big74.xlsx' → 1.4E1  |  0% ⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕   
  rp.APLSource ReadData'/tmp/big74.xlsx'        → 1.5E1  | +5% ⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕ 
```
## big99.xlsx is a 75.1MB file
```
      ]runtime -c "c.XL2APL.APLSource ReadData '/tmp/big99.xlsx'" "rp.APLSource ReadData'/tmp/big99.xlsx'"
                                                                                                         
  c.XL2APL.APLSource ReadData '/tmp/big99.xlsx' → 4.0E1  |   0% ⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕ 
  rp.APLSource ReadData'/tmp/big99.xlsx'        → 3.6E1  | -11% ⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕⎕   
```

**Comment by rikedyp** [comment]
Okay, I've had a quick look and it seems that you've made changes directly. Is anything else required on my part or shall I close this pull request and delete my fork?

**Comment by rikedyp** [comment]
Ok I've had a closer look and see that it was just the simplifcation of ReadZipFile, I'll look into re-forking and putting in .NET Core support in a couple weeks' time

**Pull request closed without merging**